### PR TITLE
Add full GitLab CI

### PR DESCRIPTION
**Pull request by fbanning**
Closes: https://github.com/tlienart/Franklin.jl/issues/362

### CODE REVIEW

File: src/templates/common/.gitlab-ci.yml [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
-        paths:
-            - public
-    only:
-        - master
+  stage: deploy
+  script:
+    - julia --project=@. -e 'import Pkg; Pkg.instantiate();
+      using NodeJS; run(`$(npm_cmd()) install highlight.js`);
+      using Franklin;
+      serve(single = true, is_final_pass = true);
```
Comment (tlienart) [inline src/templates/common/.gitlab-ci.yml:12]: this line `(serve ...)` is not needed (if you want to convince yourself, have a look at the implementation of `optimize`: https://github.com/tlienart/Franklin.jl/blob/2db712b5d4bcf28dde4b44042e88e6b1dc87f163/src/manager/post_processing.jl#L55-L58 ;-)

**Review by tlienart — COMMENTED** [review]
This is great, thank you, could you point me to a website where you have applied this so that I can quickly check that the minification + prerendering worked fine?

I expect the minification did not work because the script does not install python; it should be easy to add it with `- apt-get install -y python-dev` (could you try?)

Thanks a ton!

File: src/templates/common/.gitlab-ci.yml [hunk 2]
```diff
@@ -1,9 +1,19 @@
+image: julia:1.5
+
+before_script:
+  - apt-get -qq update; apt-get -y install git
```
Comment (tlienart) [inline src/templates/common/.gitlab-ci.yml:4]: so here probably add `- apt-get install -y python-dev`

Comment (tlienart) [reply src/templates/common/.gitlab-ci.yml:4]: possibly on the same line so `apt-get -y install git python-dev`

File: src/templates/common/.gitlab-ci.yml (hunk 1 above)
Comment (fbanning) [reply src/templates/common/.gitlab-ci.yml:12]: Didn't look at that function beforehand, seems like it does what's needed. Will remove it.

**Comment by fbanning** [comment]
I've applied both changes you mentioned above.

>  could you point me to a website where you have applied this so that I can quickly check that the minification + prerendering worked fine?

Sorry, the website I'm using this for is very simple and doesn't make any use of JS or Python features that come with Franklin. It's therefore of no use for this purpose.

You could just upload the Franklin.jl website to GitLab to test it out. Would that be feasible?

**Comment by tlienart** [comment]
As long as optimise is called then python and JS is used irrelevant of how simple your site is, the first one for minification, the latter for any math or code.

But ok I'll test this myself, thanks again for your input!

**Comment by tlienart** [comment]
@fbanning if you're interested, [this](https://github.com/tlienart/FranklinTemplates.jl/blob/master/src/templates/common/.gitlab-ci.yml) is the adjusted CI tested an all; as you can see yours was pretty much 99% there :)

Thanks again!

**Comment by fbanning** [comment]
Glad I could give something back. :)

**Pull request merged**